### ISSUE TITLE
[ELE-872] Add Databricks unity catalog to our testing

### ISSUE THREAD

**Issue by Maayan-s** [elementary-data/elementary#872]
- Users reported that anomaly detection fails
- This could unlock schema changes and dbt_columns on Databricks

<sub>[ELE-872](https://linear.app/elementary/issue/ELE-872/add-databricks-unity-catalog-to-our-testing)</sub>

**Comment by Maayan-s** [comment]
https://elementary-community.slack.com/archives/C02CTC89LAX/p1683345710056999

**Comment by Maayan-s** [comment]
@noaKurman 
I think that we actually need to have on our CI 2 instances of Databricks, one with the unity catalog enabled and one without. This is still a new feature, so we need to assume that for a while we will have users that enabled it and users that haven't (so we need to test all changes on both).

**Comment by ValdarT** [comment]
Not sure if the right place for this but we got information schema errors after upgrading the elementary dbt package from v0.7.6 to v0.7.8

> 14:41:27  Runtime Error in model filtered_information_schema_tables (models/edr/metadata_store/filtered_information_schema_tables.sql)
14:41:27    [TABLE_OR_VIEW_NOT_FOUND] The table or view `INFORMATION_SCHEMA`.`TABLES` cannot be found. Verify the spelling and correctness of the schema and catalog.

> 14:41:27  Runtime Error in model filtered_information_schema_columns (models/edr/metadata_store/filtered_information_schema_columns.sql)
14:41:27    [TABLE_OR_VIEW_NOT_FOUND] The table or view `INFORMATION_SCHEMA`.`COLUMNS` cannot be found. Verify the spelling and correctness of the schema and catalog.

> 14:41:27  Runtime Error in model dbt_columns (models/edr/dbt_artifacts/dbt_columns.sql)
14:41:27    [TABLE_OR_VIEW_NOT_FOUND] The table or view `INFORMATION_SCHEMA`.`COLUMNS` cannot be found. Verify the spelling and correctness of the schema and catalog.

We are on Databricks but NOT on Unity Catalog. Ran with latest Databricks adaptor (`dbt-databricks==1.5.2`)
